### PR TITLE
Add to the API the ability to search for an allocation based on the a…

### PR DESCRIPTION
**Pull request by death-droid**
…ssigned server id.

### CODE REVIEW

File: app/Http/Controllers/API/NodeController.php [hunk 1]
```diff
@@ -226,6 +226,33 @@ public function allocations(Request $request)
...
+     * @Get("/nodes/allocations/{id}")
+     * @Versions({"v1"})
+     * @Response(200)
+     */
+     public function allocations_view(Request $request, $id)
+     {
+        $query = Models\Allocation::where('assigned_to', $id);
+        try {
+            if (!$query->first()) {
+                throw new NotFoundHttpException('No allocation with that ID was found.');
```
Comment (DaneEveritt) [inline app/Http/Controllers/API/NodeController.php]: Shouldn't this be something along the lines of 'No allocations for that server were found.'

File: app/Http/Controllers/API/NodeController.php [hunk 2]
```diff
@@ -226,6 +226,33 @@ public function allocations(Request $request)
...
+            }
+    
+            $allocation = $query->first();
```
Comment (DaneEveritt) [inline app/Http/Controllers/API/NodeController.php]: This is just going to return a single allocation for a server, there might be multiple assigned.

File: app/Http/Routes/APIRoutes.php [hunk 3]
```diff
@@ -140,6 +140,11 @@ public function map(Router $router) {
                 'uses' => 'Pterodactyl\Http\Controllers\API\NodeController@allocations'
             ]);
 
+            $api->get('nodes/allocations/{id}', [
+                'as' => 'api.admin.nodes.allocations',
+                'uses' => 'Pterodactyl\Http\Controllers\API\NodeController@allocations_view'
```
Comment (DaneEveritt) [inline app/Http/Routes/APIRoutes.php]: Function names should follow PSR styling, `allocationsView`

File: app/Http/Controllers/API/NodeController.php [hunk 4]
```diff
@@ -226,6 +226,33 @@ public function allocations(Request $request)
...
+    /**
+     * List Node Allocation based on assigned to ID
+     *
+     * Returns a listing of the allocation for the specified server id.
+     *
+     * @Get("/nodes/allocations/{id}")
+     * @Versions({"v1"})
+     * @Response(200)
+     */
+     public function allocations_view(Request $request, $id)
```
Comment (DaneEveritt) [inline app/Http/Controllers/API/NodeController.php]: Function names should follow PSR styling, `allocationsView`

**Comment by death-droid** [comment]
Let me know if theres anything else that requires changes with it.

**Comment by DaneEveritt** [comment]
Im not sure that code actually works, have you tested it locally?

**Comment by death-droid** [comment]
Yeah i have tested it locally. Using it for my whmcs module

File: app/Http/Controllers/API/NodeController.php [hunk 5]
```diff
@@ -226,6 +226,34 @@ public function allocations(Request $request)
...
+     *
+     * Returns a listing of the allocation for the specified server id.
+     *
+     * @Get("/nodes/allocations/{id}")
+     * @Versions({"v1"})
+     * @Response(200)
+     */
+     public function allocationsView(Request $request, $id)
+     {
+        $query = Models\Allocation::where('assigned_to', $id);
```
Comment (DaneEveritt) [inline app/Http/Controllers/API/NodeController.php]: Can you do `->get()` here, and then check the count in the try block.

File: app/Http/Controllers/API/NodeController.php [hunk 6]
```diff
@@ -226,6 +226,34 @@ public function allocations(Request $request)
...
+        try {
+			
+			$allocations = $query->get();
+
+            if (!$allocations->first()) {
```
Comment (DaneEveritt) [inline app/Http/Controllers/API/NodeController.php]: Should be able to do `if(empty($query))` to accomplish the same thing, or even `if($query->count() < 1)`

**Comment by death-droid** [comment]
Ahhh yeah that's a much better way to handle it then I was handling it, i still have much to learn.
Pushed the changes

**Pull request merged**